src/main/java/org/terasology/splash/AbstractSplashScreen.java

```diff
@@ -96,19 +100,30 @@ public void close() {
 
     @Override
     public void post(String message) {
-        if (message != null) {
+        if (message == null) {
+            return;
+        }
+
+        synchronized (lock) {
             messageQueue.add(message);
```

Comment:
The actual queue is unbounded. However, messages are processed without delay as long as the queue is overfull. This ensures that all messages are seen and corresponding events are triggered.